community/cypher/cypher-compiler-3.1/src/main/scala/org/neo4j/cypher/internal/compiler/v3_1/commands/expressions/PointFunction.scala

```diff
@@ -19,18 +19,22 @@
  */
 package org.neo4j.cypher.internal.compiler.v3_1.commands.expressions
 
-import org.neo4j.cypher.internal.compiler.v3_1.{ExecutionContext, _}
 import org.neo4j.cypher.internal.compiler.v3_1.helpers.IsMap
 import org.neo4j.cypher.internal.compiler.v3_1.pipes.QueryState
 import org.neo4j.cypher.internal.compiler.v3_1.symbols.SymbolTable
+import org.neo4j.cypher.internal.compiler.v3_1.{ExecutionContext, _}
 import org.neo4j.cypher.internal.frontend.v3_1.CypherTypeException
 import org.neo4j.cypher.internal.frontend.v3_1.symbols._
 
 case class PointFunction(data: Expression) extends NullInNullOutExpression(data) {
 
   override def compute(value: Any, ctx: ExecutionContext)(implicit state: QueryState): Any = value match {
     case IsMap(mapCreator) =>
-      Points.fromMap(mapCreator(state.query))
+      if(mapCreator(state.query).exists(_._2==null)){
+        null
+      } else {
+        Points.fromMap(mapCreator(state.query))
```

Comment:
mapCreator called twice, perhaps it would be better to have Points.fromMap return null when passed maps containing any null values?